services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/application/ApplicationAttachmentController.kt

```diff
@@ -0,0 +1,187 @@
+    ): ResponseEntity<ByteArray> {
+        permissionOrThrow(applicationId, VIEW, ApplicationNotFoundException(applicationId))
+        applicationAttachmentService.getContent(applicationId, attachmentId).let {
+            (fileName, content) ->
+            return ResponseEntity.ok().headers(buildHeaders(fileName)).body(content)
+        }
+    }
+
+    @PostMapping
+    @Operation(
+        summary = "Upload attachment for application",
+        description = "File is available for downloading after successful virus scan."
+    )
+    @ApiResponses(
+        value =
+            [
+                ApiResponse(description = "Success.", responseCode = "200"),
+                ApiResponse(
+                    description = "Application not found.",
+                    responseCode = "404",
+                    content =
+                        [
+                            Content(
+                                schema =
+                                    Schema(implementation = ApplicationNotFoundException::class)
+                            )
+                        ]
+                ),
+                ApiResponse(
+                    description = "Attachment invalid.",
+                    responseCode = "400",
+                    content =
+                        [
+                            Content(
+                                schema = Schema(implementation = AttachmentUploadException::class)
+                            )
+                        ]
+                ),
+            ]
+    )
+    fun postAttachment(
+        @PathVariable applicationId: Long,
+        @RequestParam("tyyppi") tyyppi: ApplicationAttachmentType,
+        @RequestParam("liite") attachment: MultipartFile
+    ): ApplicationAttachmentMetadata {
+        permissionOrThrow(applicationId, EDIT, ApplicationNotFoundException(applicationId))
+        return applicationAttachmentService.addAttachment(applicationId, tyyppi, attachment)
+    }
+
+    @DeleteMapping("/{attachmentId}")
+    @ApiResponses(
+        value =
+            [
+                ApiResponse(description = "Delete attachment.", responseCode = "200"),
+                ApiResponse(
+                    description = "Attachment not found.",
+                    responseCode = "404",
+                    content =
+                        [
+                            Content(
+                                schema = Schema(implementation = AttachmentNotFoundException::class)
+                            )
+                        ]
+                ),
+            ]
+    )
+    fun removeAttachment(@PathVariable applicationId: Long, @PathVariable attachmentId: UUID) {
+        permissionOrThrow(applicationId, EDIT, AttachmentNotFoundException())
+        return applicationAttachmentService.deleteAttachment(applicationId, attachmentId)
+    }
+
+    fun permissionOrThrow(
+        applicationId: Long,
+        permissionCode: PermissionCode,
+        exception: RuntimeException
```

Comment:
This method checks if the user has permission for the application with the `applicationId` in the path, so it should always throw `ApplicationNotFoundException(applicationId)`.

Reply:
Removed the parameter. Throws only ApplicationNotFoundException